src/tools/compiletest/src/util.rs

```diff
@@ -211,6 +213,55 @@ pub fn logv(config: &Config, s: String) {
     }
 }
 
+/// Parses the output of `rustc --print cfg` into a [`HashMap`] containing
+/// the cfg values for each name. If the name is present with no value,
+/// it is treated as having a value of "".
```

Comment:
Does this match rustc's handling of no value cfgs? If not, is there a reason for using an empty string instead of an empty vector?

Reply:
Well, I thought that rustc is inconsistent. In some cases it has just the value, with no `=`, and in other cases it has the empty string as a value. Looking more closely, there is a semantic difference. The empty string indicates that there could be a value but isn't one, whereas no value indicates that there couldn't be a value. For example, my Mac has `target_abi=""` but also `unix`.

I standardized on the empty string because that was easier to code into `cfg_add`. There's no way to match on the empty string specifically, and no point in adding one, so it's pretty irrelevant for our purposes.  I can change it if you want. I do like the elegance of this code though... :)